test.py

```diff
@@ -30,13 +31,14 @@
 class NodeFactory(object):
     """A factory to setup and start `lightningd` daemons.
     """
-    def __init__(self, testname, executor, btc, btcd):
+    def __init__(self, testname, executor, btc, btcd, electrumx):
```

Comment:
Oh I see, we're not really injecting fixtures here, we're instantiating a `NodeFactory`. Let's see if we can make this a bit more lightweight somehow.

Reply:
Maybe having a reference to the `NodeFactory` and pulling in the `electrumx` dependency on demand might be the key here.